docker-for-aws/persistent-data-volumes.md

```diff
@@ -22,44 +32,86 @@ f416c95c0dcc        cloudstor:aws               cloud storage plugin for Docker
 
-### Use a unique volume per task:
+The only option available for EFS currently is `perfmode`. If high IO throughput is desired, please set the `perfmode` volume-opt to `maxio`. Example:
 
 ```bash
 {% raw %}
-docker service create --replicas 5 --name ping2 \
-    --mount type=volume,volume-driver=cloudstor:aws,source={{.Service.Name}}-{{.Task.Slot}}-vol,destination=/mydata \
+docker service create --replicas 5 --name ping3 \
+    --mount type=volume,volume-driver=docker4x/cloudstor:aws-v{{ edition_version }},source={{.Service.Name}}-{{.Task.Slot}}-vol5,destination=/mydata,volume-opt=perfmode=maxio \
     alpine ping docker.com
 {% endraw %}
 ```
 
-Here the templatized notation is used to indicate to Docker Swarm that a unique volume be created and mounted for each replica/task of the service `ping2`. After initial creation of the volumes corresponding to the tasks they are attached to (in the nodes the tasks are scheduled in), if a task is rescheduled on a different node, Docker Swarm will interact with the Cloudstor plugin to create and mount the volume corresponding to the task on the node the task got scheduled on. It's highly recommended that you use the `.Task.Slot` template to make sure task N always gets access to vol N no matter which node it is executing on/scheduled to.
-
-In the above example, each task has it's own volume mounted at `/mydata/` and the files under there are unique to the task mounting the volume.
+It is also possible to create `shared` Cloudstor volumes using the `docker volume create` CLI. For example:
 
-### List or remove volumes created by Cloudstor
+```bash
+{% raw %}
+docker volume create -d "cloudstor:aws" --opt backing=shared mysharedvol1
+{% endraw %}
+```
 
-You can use `docker volume ls` to enumerate all volumes created on a node including those backed by Cloudstor. Note that if a swarm service task starts off in a node and has a Cloudstor volume associated and later gets rescheduled to a different node, `docker volume ls` in the initial node will continue to list the Cloudstor volume that was created for the task that no longer executes on the node although the volume is mounted elsewhere. Do NOT prune/rm the volumes that gets enumerated on a node without any tasks associated since these actions will result in data loss if the same volume is mounted in another node (i.e. the volume shows up in the `docker volume ls` output on another node in the swarm). We can try to detect this and block/handle in post-Beta.
+### Use a unique volume per task (using EBS):
 
-### Configure IO performance
+A unique `local` Cloudstor volume backed by a specified type of EBS can be created and mounted for each task in a swarm service using a templatized notation with the `docker service create` CLI. Creation of new EBS volumes typically takes a few minutes.  The volume options (besides `backing=local` indicating `local` EBS backed volumes) are:
```

Comment:
what does the container see while the ebs volume is provisioning? or is the task not scheduled before the volume is ready? (if that's the case, maybe make a note of that)